docs/design/generic-enhanced-purls.md

```diff
@@ -0,0 +1,233 @@
+# Support for different purl types in the generic artifact fetcher
+
+The generic artifact package manager is being added to Cachi2 as a means for users to introduce files that do not belong to traditional package manager ecosystems (e.g. pip, npm, golang) to their hermetic container builds. Since Cachi2 does not have any extra information about the file that's being fetched, the purls are always reported as [pkg:generic](https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#generic).
```

Comment:
Is this a generic purl type and/or a generic property, i.e.
```
    "properties": [
        {
            "name": "cachi2:found_by",
            "value": "cachi2:generic"
        }
```

Reply:
In this case, a generic purl type. The property is a way to inform which backend/package manager in Cachi2 produced that SBOM component.

Reply:
Would it be possible to add an example purl that Cachi2 might generate with an input that could produce it?

Reply:
Do you need an example different from the ones in https://github.com/containerbuildsystem/cachi2/pull/692/files#diff-aadc5e0bf6d8c1eefd2742bad409f3f2fb7a684aa7cbe0d47ebfb752e07d9fd6R31?

Reply:
I guess it is sufficient. I cannot remember why I was asking specifically. This can be resolved.